client/lib/i18n-utils/switch-locale.js

```diff
@@ -521,10 +521,13 @@ export function switchWebpackCSS( isRTL ) {
 		}
 
 		const newLink = await loadCSS( newHref, currentLink );
-		newLink.setAttribute( 'data-webpack', true );
-
-		if ( currentLink.parentElement ) {
-			currentLink.parentElement.removeChild( currentLink );
+		// After the CSS files load the RTL state might have changed
+		// This is a double-check to ensure the value of isRTL
+		if ( i18n.isRtl() === isRTL && newLink ) {
```

Comment:
As we have double-checked in `loadCSS`, I think we don't need to check again. We just need to check `newLink`. If it is not null, then we have to set the `data-webpack` attribute and remove the current link.

Reply:
Because it is a race condition issue, I wonder if this check would also help in case the RTL value changes **after** the CSS file loads. The check within `loadCSS` happens **before** the file loads

Reply:
hmmm...I don't think the RTL value will change after the `loadCSS` function is resolved. We can add this check later if the current solution won't work. What do you think?

Reply:
I meant during the CSS files are loading, not exactly after `loadCSS` but I agree with testing the solution without that check first 👍  I have just pushed the update.